test/jsonapi/plugs/deserializer_test.exs

```diff
@@ -0,0 +1,183 @@
+defmodule JSONAPI.DeserializerTest do
+  use ExUnit.Case
+  use Plug.Test
+
+  defmodule ExamplePlug do
+    use Plug.Builder
+    plug Plug.Parsers, parsers: [:json], json_decoder: Jason
+    plug JSONAPI.Deserializer
+    plug :return
+
+    def return(conn, _opts) do
+      send_resp(conn, 200, "success")
+    end
+  end
+
+  @ct "application/vnd.api+json"
```

Comment:
You may just want to reference, or alias, `JSONAPI.mime_type` so that we don't spread this string throughout the code.